src/sentry/replays/endpoints/project_replay_viewed_by.py

```diff
@@ -103,9 +105,33 @@ def post(self, request: Request, project: Project, replay_id: str) -> Response:
         except ValueError:
             return Response(status=404)
 
-        message = viewed_event(project.id, replay_id, request.user.id)
-        publish_replay_event(message, is_async=False)
+        # make a query to avoid overwriting the `finished_at` column
+        filter_params = self.get_filter_params(request, project, date_filter_optional=False)
+        finished_at_response = execute_query(
+            query=make_full_aggregation_query(
+                fields=["finished_at"],
+                replay_ids=[replay_id],
+                project_ids=[project.id],
+                period_start=filter_params["start"],
+                period_end=filter_params["end"],
+                request_user_id=request.user.id,
+            ),
+            tenant_id={"organization_id": project.organization.id} if project.organization else {},
```

Comment:
i don't think project.organization would ever not exist?

Reply:
Not sure.. I copied this pattern from other functions in usecases/query.py, might be good to be safe

Reply:
> might be good to be safe

https://github.com/getsentry/sentry/blob/8666d0c9f60be94988545522e1ff74aa7c0002f5/src/sentry/models/project.py#L229

organization is a required field on project, so i'm not sure this is doing much for us -- would also be curious what's up with usecases/query.py as well.

Reply:
Ah ok.. thanks for finding a reference. I can follow up in a pr to refactor query code?

Reply:
nbd, if you have a moment, if not no worries